flow/connectors/clickhouse/clickhouse.go

```diff
@@ -102,34 +109,75 @@ func NewClickhouseConnector(
 		return nil, err
 	}
 
-	var clickhouseS3Creds *utils.ClickhouseS3Credentials
-	// Get user provided S3 credentials
-	clickhouseS3Creds = &utils.ClickhouseS3Credentials{
-		AccessKeyID:     config.AccessKeyId,
-		SecretAccessKey: config.SecretAccessKey,
-		Region:          config.Region,
-		BucketPath:      config.S3Path,
+	credentialsProvider, err := utils.GetAWSCredentialsProvider(ctx, "clickhouse", utils.PeerAWSCredentials{
+		Credentials: aws.Credentials{
+			AccessKeyID:     config.AccessKeyId,
+			SecretAccessKey: config.SecretAccessKey,
+		},
+		EndpointUrl: nil,
+		Region:      config.Region,
+	})
+	if err != nil {
+		return nil, err
 	}
 
-	if clickhouseS3Creds.AccessKeyID == "" &&
-		clickhouseS3Creds.SecretAccessKey == "" && clickhouseS3Creds.Region == "" &&
-		clickhouseS3Creds.BucketPath == "" {
+	awsBucketPath := config.S3Path
+
+	if awsBucketPath == "" {
 		deploymentUID := peerdbenv.PeerDBDeploymentUID()
 		flowName, _ := ctx.Value(shared.FlowNameKey).(string)
 		bucketPathSuffix := fmt.Sprintf("%s/%s",
 			url.PathEscape(deploymentUID), url.PathEscape(flowName))
-
 		// Fallback: Get S3 credentials from environment
-		clickhouseS3Creds = utils.GetClickhouseAWSSecrets(bucketPathSuffix)
+		awsBucketName := os.Getenv("PEERDB_CLICKHOUSE_AWS_S3_BUCKET_NAME")
+		if awsBucketName == "" {
+			return nil, errors.New("PeerDB Clickhouse Bucket Name not set")
+		}
+
+		awsBucketPath = fmt.Sprintf("s3://%s/%s", awsBucketName, bucketPathSuffix)
+	}
+	clickHouseS3CredentialsNew := utils.ClickHouseS3Credentials{
+		Provider:   credentialsProvider,
+		BucketPath: awsBucketPath,
+	}
+	credentials, err := credentialsProvider.Retrieve(ctx)
+	if err != nil {
+		return nil, err
+	}
+	if credentials.AWS.SessionToken != "" {
+		minSupportedClickhouseVersion := "v24.3.1"
```

Comment:
why is 24.3.1 minimum version? s3 express one zone? Comment should specify rationale for version requirement

Reply:
This is the minimum version of Clickhouse that actually supports session token (https://github.com/ClickHouse/ClickHouse/issues/61230)

Added a comment